pkg/controller/vitesscluster/reconcile_vtadmin.go

```diff
@@ -0,0 +1,373 @@
+/*
+Copyright 2022 PlanetScale Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vitesscluster
+
+import (
+	"context"
+	"fmt"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	planetscalev2 "planetscale.dev/vitess-operator/pkg/apis/planetscale/v2"
+	"planetscale.dev/vitess-operator/pkg/operator/conditions"
+	"planetscale.dev/vitess-operator/pkg/operator/reconciler"
+	"planetscale.dev/vitess-operator/pkg/operator/results"
+	"planetscale.dev/vitess-operator/pkg/operator/update"
+	"planetscale.dev/vitess-operator/pkg/operator/vitesscell"
+	"planetscale.dev/vitess-operator/pkg/operator/vtadmin"
+	"planetscale.dev/vitess-operator/pkg/operator/vtctld"
+	"planetscale.dev/vitess-operator/pkg/operator/vtgate"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileVitessCluster) reconcileVtadmin(ctx context.Context, vt *planetscalev2.VitessCluster) (reconcile.Result, error) {
+	resultBuilder := results.Builder{}
+	// Do not deploy vtadmin if not configured
+	if vt.Spec.VtAdmin == nil {
+		return resultBuilder.Result()
+	}
+
+	// Some checks to validate user input
+	if len(vt.Spec.Images.Vtadmin) == 0 {
+		log.Error("Not deploying vtadmin since image is unspecified")
+		return resultBuilder.Result()
+	}
+
+	if len(vt.Spec.VtAdmin.APIAddresses) == 0 {
+		log.Errorf("Not deploying vtadmin since api addresses field is not specified. Atleast 1 value is required")
+	}
+
+	if len(vt.Spec.VtAdmin.APIAddresses) != 1 && len(vt.Spec.VtAdmin.APIAddresses) != len(vt.Spec.VtAdmin.Cells) {
+		log.Errorf("Not deploying vtadmin since api addresses field doesn't align with cells field")
+	}
```

Comment:
These are user input validations. There exist 3 - 
- The VTAdmin image should be specifically provided. We do not add a default for `vitess:lite` since that won't work
- There should be at least 1 APIAddress provided
- If more than 1 APIAddresses are provided, then the length of this list should match the length of the cells list.

Reply:
That makes sense to me